test/e2e/bootstrap/bootstrap_signer.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	bootstrapapi "k8s.io/kubernetes/pkg/bootstrap/api"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	TokenIDBytes     = 3
+	TokenSecretBytes = 8
+)
+
+var _ = framework.KubeDescribe("[Feature:BootstrapSigner]", func() {
+
+	var c clientset.Interface
+
+	f := framework.NewDefaultFramework("bootstrap-signer")
+
+	BeforeEach(func() {
+		c = f.ClientSet
+	})
+
+	It("should sign the new added bootstrap tokens", func() {
+		By("create a new bootstrap token secret")
+		tokenId, err := generateTokenId()
+		Expect(err).NotTo(HaveOccurred())
+		secret := newTokenSecret(tokenId, "tokenSecret")
+		_, err = c.CoreV1().Secrets(metav1.NamespaceSystem).Create(secret)
+		defer func() {
+			By("delete the bootstrap token secret")
+			err := c.CoreV1().Secrets(metav1.NamespaceSystem).Delete(bootstrapapi.BootstrapTokenSecretPrefix+tokenId, &metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("wait for the bootstrap token secret be signed")
+		err = framework.WaitforSignedBootStrapToken(c, tokenId)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should resign the bootstrap tokens when the clusterInfo ConfigMap updated [Serial][Disruptive]", func() {
```

Comment:
This will modify the main cluster info configmap.  If things fail out the cluster could be slightly messed up.